core/metamodel.class.php

```diff
@@ -4180,18 +4180,18 @@ public static function PrepareQueryArguments($aArgs, $aMoreArgs = array(), $aExp
 				}
 				else
 				{
-					if (is_scalar($value))
-					{
+					if (is_scalar($value)) {
 						$aScalarArgs[$sArgName] = (string)$value;
-					}
-					elseif (is_null($value))
-					{
+					} elseif (is_null($value)) {
 						$aScalarArgs[$sArgName] = null;
-					}
-					elseif (is_array($value))
-					{
+					} elseif (is_array($value)) {
 						$aScalarArgs[$sArgName] = $value;
+					} else {
+						if (is_object($value) && method_exists($value, 'GetValueForQuery')) {
```

Comment:
```suggestion
						if ($value instanceof iOrmset) {
```
https://3v4l.org/RRMSR

Reply:
This was my first idea, but this fix is ​​for Teemip first. I waited to meet xtophe and validate that it's ok for him to lose 3.0 compatibility in his extension. I can now make this change.

Reply:
How will this break 3.0 compatibility?